main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -639,11 +639,48 @@ object Parser2 {
   private val NAME_PREDICATE: Set[TokenKind] = Set(TokenKind.NameUpperCase)
 
   /**
-    * Consumes a token if kind is in `kinds`.
-    * If `allowQualified` is passed also consume subsequent dot-separated tokens with kind in `kinds`.
-    * If the found kind is in the tail set, then no further dot separated tokens will be consume
+    *   - `KIND` is consumed and produces [[TreeKind.Ident]]
+    *   - [[TokenKind.isFirstDecl]] and [[TokenKind.isKeyword]] (not consumed)
+    *     produces [[TreeKind.ErrorTree]]
+    *   - `ANY` is consumed and produces [[TreeKind.ErrorTree]]
     */
-  private def name(kinds: Set[TokenKind], tail: Set[TokenKind] = Set(TokenKind.NameLowerCase), allowQualified: Boolean = false, context: SyntacticContext)(implicit s: State): Mark.Closed = {
+  private def name(kinds: Set[TokenKind], context: SyntacticContext)(implicit s: State): Mark.Closed = {
+    nameGeneric(kinds, tail = Set.empty, allowQualified = false, context)
+  }
+
+  /**
+    *   - `KIND` is consumed and produces [[TreeKind.Ident]]
+    *   - `KIND.KIND.KIND` is consumed and produces [[TreeKind.QName]]([[TreeKind.Ident]]..)
+    *   - `KIND.KIND.` is consumed and produces
+    *     [[TreeKind.QName]]([[TreeKind.Ident]].., [[TreeKind.ErrorTree]])
+    *   - `KIND. KIND.KIND` is consumed and produces
+    *     [[TreeKind.QName]]([[TreeKind.Ident]].., [[TreeKind.ErrorTree]], [[TreeKind.Ident]]..)
+    *   - [[TokenKind.isFirstDecl]] and [[TokenKind.isKeyword]] (not consumed)
+    *     produces [[TreeKind.ErrorTree]]
+    *   - `ANY` is consumed and produces [[TreeKind.ErrorTree]]
+    *
```

Comment:
It is really hard to understand the above.

Reply:
I am trying to describe 1) what is parsed 2) what is consumed 3) what tree is produced 4) are errors actively reported

And for rules like this, it doesn't just do one thing, but chooses among several different options

Reply:
But what does KIND.KIND.KIND even mean? And by Kind its not the kind of a type or?

Reply:
I agree, it's confusing. I think it would be a bit easier if we had example strings instead of `KIND`

Reply:
@magnus-madsen but do you agree with the change, ignoring the comment? I can always make the comment as vague as it was before